mlir/lib/Dialect/MQTOpt/Transforms/ElidePermutationsPattern.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2023 - 2025 Chair for Design Automation, TUM
+ * Copyright (c) 2025 Munich Quantum Software Company GmbH
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: MIT
+ *
+ * Licensed under the MIT License
+ */
+
+#include "mlir/Dialect/MQTOpt/IR/MQTOptDialect.h"
+#include "mlir/Dialect/MQTOpt/Transforms/Passes.h"
+
+#include <llvm/ADT/STLExtras.h>
+#include <mlir/IR/MLIRContext.h>
+#include <mlir/IR/PatternMatch.h>
+#include <mlir/IR/Value.h>
+#include <mlir/IR/ValueRange.h>
+#include <mlir/Support/LLVM.h>
+#include <mlir/Support/LogicalResult.h>
+#include <vector>
+
+namespace mqt::ir::opt {
+
+/**
+ * @brief This pattern attempts to remove SWAP gates by re-ordering qubits.
+ */
+struct ElidePermutationsPattern final : mlir::OpRewritePattern<SWAPOp> {
+
+  explicit ElidePermutationsPattern(mlir::MLIRContext* context)
+      : OpRewritePattern(context) {}
+
+  mlir::LogicalResult
+  matchAndRewrite(SWAPOp op, mlir::PatternRewriter& rewriter) const override {
+    const bool isControlledSwap = !op.getAllCtrlInQubits().empty();
+    if (isControlledSwap) {
+      return mlir::failure();
+    }
+
+    auto inQubits = op.getInQubits();
+
+    std::vector<mlir::Value> swappedQubits;
+    for (auto&& reversed : llvm::reverse(inQubits)) {
+      swappedQubits.push_back(reversed);
+    }
+
+    rewriter.replaceAllOpUsesWith(op, swappedQubits);
```

Comment:
Given how it is compile-time known that the SWAP will always have two targets, I believe this logic could be even simpler. Something like
```suggestion
    rewriter.replaceAllOpUsesWith(op, {inQubits[1], inQubits[0]});
```
should just work and achieve the same result